dashboard/app/controllers/admin_reports_controller.rb

```diff
@@ -259,6 +259,24 @@ def hoc_signups
     end
   end
 
+  def retention
+    require 'cdo/properties'
+    SeamlessDatabasePool.use_persistent_read_connection do
+      @scripts = params[:scripts_ids].present? ?
+        params[:script_ids].split(',').map(&:to_i) :
+        [1, 17, 18, 19, 23]  # Default to the CSF scripts.
+      # Get the cached retention_stats from the DB, trimming those stats to only the requested
+      # scripts.
+      @retention_stats = Properties.get(:retention_stats).each_pair do |key, key_data|
+        key_data.delete_if{|script_id, script_data| !@scripts.include? script_id.to_i}
```

Comment:
Note that the select would be a ruby select, not a DB select.

Is
@retention_stats = {}
Properties.get(:retention_stats).each_pair do |key, key_data|
  @retention_stats[key] = key_data.select!{|script_id, script_data| @scripts.include? script_id.to_i}
end

preferred? Happy to change it to this if it is.